app/org/sagebionetworks/bridge/play/controllers/AuthenticationController.java

```diff
@@ -52,10 +52,36 @@ public Result emailSignIn() {
         CriteriaContext context = getCriteriaContext(study.getStudyIdentifier());
         
         UserSession session = authenticationService.emailSignIn(context, signInRequest);
-        
+        logAuthenticationSuccess(session);
+
         return okResult(UserSessionInfo.toJSON(session));
     }
 
+    public Result requestPhoneSignIn() {
+        SignIn signInRequest = parseJson(request(), SignIn.class);
+        
+        authenticationService.requestPhoneSignIn(signInRequest);
+
+        return acceptedResult("Message sent.");
+    }
+
+    public Result phoneSignIn() {
+        SignIn signInRequest = parseJson(request(), SignIn.class);
+
+        if (isBlank(signInRequest.getStudyId())) {
+            throw new BadRequestException("Study identifier is required.");
+        }
+        Study study = studyService.getStudy(signInRequest.getStudyId());
+        verifySupportedVersionOrThrowException(study);
```

Comment:
Is it worth adding a unit test that phoneSignIn() checks the study shutoff switch?

Reply:
Added